pkg/dynamodb/dynamo_api.go

```diff
@@ -0,0 +1,419 @@
+package dynamodb
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/aws/aws-sdk-go-v2/feature/dynamodb/attributevalue"
+
+	"time"
+
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/feature/dynamodb/expression"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb/types"
+)
+
+type Keys struct {
+	PartitionKey string `dynamodbav:"pk"`
+	SortKey      string `dynamodbav:"sk"`
+}
+
+type GetItemParams struct {
+	KeyCondition   Keys
+	ConsistentRead bool
+}
+
+type QueryItemsParams struct {
+	ProjectionKeys   []string
+	FilterParamsList []ItemParams
+	PartitionKey     string
+	SortKey          ItemParams
+	ConsistentRead   bool
+}
+
+type ItemParams struct {
+	KeyName       string
+	ConditionName string
+	Value         interface{}
+}
+
+type DeleteItemParams struct {
+	KeyCondition Keys
+}
+
+type BatchGetItemParams struct {
+	KeyCondition   []Keys
+	ProjectionKeys []string
+	ConsistentRead bool
+}
+
+func (dynamoClient *DynamoClient) GetItem(ctx context.Context, inputRequest GetItemParams, outputItem interface{}) (interface{}, error) {
+	row, err := marshalMap(inputRequest.KeyCondition)
+
+	if err != nil {
+		return nil, err
+	}
+
+	input := &dynamodb.GetItemInput{
+		TableName:      aws.String(dynamoClient.Table),
+		Key:            row,
+		ConsistentRead: &inputRequest.ConsistentRead,
+	}
+	result, err := dynamoClient.getItem(ctx, input)
+	if err != nil {
+		return nil, err
+	}
+
+	if outputItem == nil {
+		return result.Item, err
+	}
+	err = unmarshalMap(result.Item, outputItem)
+
+	return outputItem, err
+}
+
+func (dynamoClient *DynamoClient) PutItem(ctx context.Context, values interface{}) error {
+	row, err := marshalMap(values)
+
+	if err != nil {
+		return err
+	}
+
+	input := &dynamodb.PutItemInput{
+		Item:                row,
+		TableName:           aws.String(dynamoClient.Table),
+		ConditionExpression: aws.String("attribute_not_exists(partitionKey)"),
+	}
+
+	_, err = dynamoClient.putItem(ctx, input)
+
+	return err
+}
+
+type UpdateItemParams struct {
+	ValuesToBeUpdated     map[string]interface{}
+	ValuesToBeRemoved     []string
+	KeyCondition          Keys
+	ValuesToBeIncremented []string
+	ValuesToBeAppended    map[string]interface{}
+	ConditionalParams     []ItemParams
+}
+
+func (dynamoClient *DynamoClient) UpdateItem(ctx context.Context, inputRequest *UpdateItemParams, outputItem interface{}) error {
+	expr, err := makeUpdateExpression(inputRequest)
+
+	if err != nil {
+		return err
```

Comment:
**Suggestion:** In the `UpdateItem` method, ensure that `inputRequest` is not nil before proceeding with the update operation to prevent potential nil pointer dereference errors. [possible issue, importance: 9]
```suggestion
if inputRequest == nil {
    return errors.New("inputRequest cannot be nil")
}
expr, err := makeUpdateExpression(inputRequest)

if err != nil {
    return err
}
```